code/game/objects/items/weapons/material/knives.dm

```diff
@@ -25,6 +25,26 @@
 			M = user
 		return eyestab(M,user)
 
+/obj/item/weapon/material/knife/verb/extract_shrapnel(var/mob/living/carbon/human/H)
+	set name = "Extract Shrapnel"
+	set category = "Object"
+	set src in usr
+
+	for(var/obj/item/weapon/material/shard/shrapnel/S in H.contents)
+		visible_message("<span class='notice'>[usr] starts carefully digging out some of the shrapnel in [H == usr ? "themselves" : H]...</span>")
+		to_chat(H, "<font size=3><span class='danger'>It burns!</span></font>")
+		if(do_after(H, 70))
+			S.forceMove(H.loc)
+			log_and_message_admins("has extracted shrapnel out of [H.key]/[H]", H)
+		H.apply_damage(30, HALLOSS)
+		if(prob(25))
+			var/obj/item/organ/external/affecting = H.get_organ(H.zone_sel.selecting)
+			if(affecting)
+				var/datum/wound/internal_bleeding/I = new(15)

```

Comment:
Maybe add some varied feedback here to indicate issues? Specially since internal bleeding is horribly sneaky otherwise.